index.js

```diff
@@ -582,7 +583,18 @@ function adapter(uri, opts) {
       // if there is no response for x second, return result
       var timeout = setTimeout(function() {
         var request = self.requests[requestid];
-        if (fn) process.nextTick(fn.bind(null, new Error(`timeout reached while waiting for clients response, rooms: ${rooms}, err: ${self.requests[requestid].errMessage}, clients: ${Object.keys(request.clients)}`), Object.keys(request.clients)));
+
+        // 1. atleast one node responded (inclusive of self)
+        // 2. msgCount is 1 less than expected (happens in case of a scaledown)
+        if (request.msgCount > 0 && request.numsub == (request.msgCount + 1) && retryCount <= maxRetryCount) {
+          // add a retry here
+          self.emit('error', `Retrying client request ${retryCount} ${JSON.stringify(request)}`);
```

Comment:
What are these emits for? logging? @ShubhankarS

Reply:
yes. to escalate main logger